docs/golang_extension.md

```diff
@@ -0,0 +1,93 @@
+# Golang (plugin) extensions
+
+Gohan supports extensions written in golang and imported as shared libraries.
+Golang plugins are supported starting from version 1.8 of golang. Note that
+there exist a bug [1] which will be fixed in version 1.9.
+
+## Why golang extensions?
+
+One of the main reasons to choose a golang extensions over a javascript ones is the
+performance. Golang plugins are compiled to native code (not interpreted as javascript
```

Comment:
I also added a table to compare pros & cons of different types of extensions to see a clear difference